roofit/roofitcore/inc/RooRealVar.h

```diff
@@ -164,7 +164,7 @@ class RooRealVar : public RooAbsRealLValue {
   
   std::shared_ptr<RooRealVarSharedProperties> _sharedProp; //! Shared binnings associated with this instance
 
-  ClassDef(RooRealVar,7) // Real-valued variable
+  ClassDef(RooRealVar,8) // Real-valued variable
```

Comment:
This is alarming.
When did this happen?
Does it need to be backported?
Why do you see it only now?

You should get this PR merged (nag Axel and everyone who has a failing test under their code ownership about it, I'll rebase it for you):
https://github.com/root-project/root/pull/5591

Reply:
I have no idea why it happened. As you also saw in the other PR, I had the same problem when I changed the RooListProxy and RooSetProxy classes:
https://github.com/root-project/root/pull/8728#discussion_r680832877

I'll ask on Mattermost, maybe one of the I/O experts can help me.

Reply:
I can explain, actually:
- You presumably changed the pointer member for a `unique_ptr`.
- That's commendable, but schema evolution must be taken care of.
- Since we are never testing whether gtests trigger warnings such as the one above, nobody notices.

What you should do now:
- Find when the change was introduced.
- Make sure that the increment of the class version was backported to released ROOTs if necessary.
- Pressure everybody so #5591 can be merged ASAP.

Reply:
Thanks for the input, but this doesn't explain why the warning only appeared now after changing seeminlgy unrealed classes. I master there is no warning, only in master+this PR minus the last commit.

The change to `unique_ptr` was introduced by you quite some time ago, going along with incrementing the class version:
https://github.com/root-project/root/commit/2ca7f7910f720b748a01dc28ed6d981300a1acc5#diff-6614cb454d92121edef4202ab96f535572bd49cdfdf6aa155b04f9ce41b35ede

So I'm still just as confused as before about why the warning only appears now...

Reply:
In fact, the changes to `RooAbsArg.h` and `RooAbsArg.cxx` alone are sufficient to trigger the Warning in `testProxiesAndCategories`, which is not present with ROOT master:
```
Warning in <TStreamerInfo::BuildCheck>: 
   The StreamerInfo of class RooRealVar read from file categories_v622.root
   has the same version (=7) as the active class but a different checksum.
   You should update the version to ClassDef(RooRealVar,8).
   Do not try to write objects with the current class definition,
   the files will not be readable.

Warning in <TStreamerInfo::CompareContent>: The following data member of
the on-file layout version 7 of class 'RooRealVar' differs from 
the in-memory layout version 7:
   RooAbsBinning _binning; //
vs
   unique_ptr<RooAbsBinning,default_delete<RooAbsBinning> > _binning; //
```